obspy/core/util/testing.py

```diff
@@ -26,10 +26,9 @@ def compare_xml_strings(doc1, doc2):
     """
     Simple helper function to compare two XML strings.
 
-    :type doc1: str
-    :type doc2: str
+    :type doc1: bytes or str
+    :type doc2: bytes or str
```

Comment:
Still kind of a compatibility hack; we should make it one type only and remove everything in the `try`.

Reply:
Yes, it was quick and dirty; the callers use the function in both ways.

Reply:
Turns out, it some calls the first string was utf-8 and the second binary.